Sources/ApolloAPI/SchemaConfiguration.swift

```diff
@@ -1,38 +1,15 @@
-/// A protocol that a generated GraphQL schema should conform to.
+/// A protocol for an object used to provide custom configuration for a generated GraphQL schema.
 ///
-/// The generated schema is the source of information about the generated types in the schema.
-/// It is used to map each object in a `GraphQLResponse` to the ``Object`` type representing the
-/// response object.
-///
-/// The generated schema also provides an entry point for customizing the cache key resolution
+/// A ``SchemaConfiguration`` provides an entry point for customizing the cache key resolution
 /// for the types in the schema, which is used by `NormalizedCache` mechanisms.
-public protocol SchemaMetadata {
-
-  static var configuration: SchemaConfiguration.Type { get }
-
-  /// Maps each object in a `GraphQLResponse` to the ``Object`` type representing the
-  /// response object.
-  ///
-  /// > Note: This function will be generated by the code generation engine. You should never
-  /// alter the generated implementation or implement this function manually.
-  ///
-  /// - Parameter typename: The value of the `__typename` field of the response object.
-  /// - Returns: An ``Object`` type representing the response object if the type is known to the
-  /// schema. If the schema does not include a known ``Object`` with the given ``Object/typename``,
-  /// returns `nil`.
-  static func objectType(forTypename typename: String) -> Object?
-}
-
 public protocol SchemaConfiguration {
   /// The entry point for configuring the cache key resolution
   /// for the types in the schema, which is used by `NormalizedCache` mechanisms.
   ///
-  /// The default implementation always returns `nil`, disabling all cache normalization.
+  /// The default generated implementation always returns `nil`, disabling all cache normalization.
```

Comment:
I wonder if it's worth mentioning here that disabling normalization does not mean no caching. Something simple like `...disabling all cache normalization. Responses will be cached by query path instead.`